File: web-client/js/repository/SurveyRepository.js
import {SurveyStation} from '../model/SurveyStation.js';
import {Measurement} from '../model/Measurement.js';
/**
 * Provides methods for working with a collection of survey stations
 */
export class SurveyRepository {
    #surveyStations;
    
    constructor() {
        this.#surveyStations = new Array();
    }

    /**
     * Clears repository
     */
    clearAll() {
        this.#surveyStations = new Array();
    }

    /**
     * Gets size of survey repository
     * @returns {number}
     */
    size() {
        return this.#surveyStations.length;
    }

    /**
     * Gets the number of measurements at the specified station
     * @param {number} indexStation 
     * @returns {number}
     */
    measurementSize(indexStation) {
        if(indexStation >= this.#surveyStations.length) {
            return null;
        }
        let surveyStation = this.#surveyStations.at(indexStation);
        return surveyStation.measurements.length;
    }

    /**
     * Appends new SurveyStation to the end of repository
     */
    addNewStation () {
        let surveyStation = new SurveyStation();
        this.#surveyStations.push(surveyStation);
    }

    /**
     * Inserts new SurveyStation to need position in repository
     * @param {number} index position new element at repository
     */
    insertNewStation(indexStation) {
        if(indexStation <= this.#surveyStations.length) {
            let surveyStation = new SurveyStation();
            this.#surveyStations.splice(indexStation, 0, surveyStation);
        }
    }

    /**
     * Removes element of repository from need position
     * @param {number} indexStation location of the repository item being deleted
     */
    removeStation(indexStation) {
        if(indexStation < this.#surveyStations.length) {
            this.#surveyStations.splice(indexStation,1);
        }
    }

    /**
     * Adds a new empty instance of Measurement to the end of the station measurement collection with the specified index
     * @param {number} indexStation the index of the station to which the new measurement is being added
     */
    addNewMeasurement(indexStation) {
        if(indexStation < this.#surveyStations.length) {
            let measurement = new Measurement();
            let suveyStation = this.#surveyStations.at(indexStation);
            suveyStation.measurements.push(measurement);
        }
    }

    /**
     * Adds a new empty measurement to the station's measurement collection with the specified index at the specified position
     * @param {number} indexStation surveyStation index
     * @param {number} indexMeasurement position of new measurement
     */
    insertNewMeasurement(indexStation, indexMeasurement) {
        if(indexStation < this.#surveyStations.length) {
            let surveyStation = this.#surveyStations.at(indexStation);
            if(indexMeasurement <= surveyStation.measurements.length) {
                let measurement = new Measurement();
                surveyStation.measurements.splice(indexMeasurement, 0, measurement);
            }
        }
    }

    /**
     * Deletes an item from the station's measurement collection from the specified position
     * @param {number} indexStation 
     * @param {number} indexMeasurement 
     */
    removeMeasurement(indexStation, indexMeasurement) {
        if(indexStation < this.#surveyStations.length) {
            let surveyStation = this.#surveyStations.at(indexStation);
            if(indexMeasurement < surveyStation.measurements.length) {
                surveyStation.measurements.splice(indexMeasurement, 1);
            }
        }
    }

    /**
     * Gets the name of the station with the specified index
     * @param {number} indexStation 
     * @returns {string}
     */
    getStationName(indexStation) {
        if(indexStation >= this.#surveyStations.length) {
            return null;
        }
        let surveyStation = this.#surveyStations.at(indexStation);
        return surveyStation.stationName;
    }

    /**
     * Saves the specified station name with the specified index
     * @param {number} indexStation 
     * @param {string} stationName 
     */
    saveStationName(indexStation, stationName) {
        if(indexStation < this.#surveyStations.length) {
            let surveyStation = this.#surveyStations.at(indexStation);
            surveyStation.stationName = stationName;
        }
    }

    /**
     * Gets the coordinate X of the station with the specified index in metters
     * @param {number} indexStation 
     * @returns {string}
     */
    getStationX(indexStation) {
        if(indexStation >= this.#surveyStations.length) {
            return null;
        }
        let surveyStation = this.#surveyStations.at(indexStation);
        return surveyStation.stationX;
    }

    /**
     * Saves the specified coordinate X for station with the specified index in metters
     * @param {number} indexStation 
     * @param {string} stationX 
     */
    saveStationX(indexStation, stationX) {
        if(indexStation < this.#surveyStations.length) {
            let surveyStation = this.#surveyStations.at(indexStation);
            surveyStation.stationX = stationX;
        }
    }    
    
    /**
     * Gets the coordinate X of the station with the specified index in metters
     * @param {number} indexStation 
     * @returns {string}
     */
    getStationY(indexStation) {
        if(indexStation >= this.#surveyStations.length) {
            return null;
        }
        let surveyStation = this.#surveyStations.at(indexStation);
        return surveyStation.stationY;
    }

    /**
     * Saves the specified coordinate Y for station with the specified index in metters
     * @param {number} indexStation 
     * @param {string} stationY 
     */
    saveStationY(indexStation, stationY) {
        if(indexStation < this.#surveyStations.length) {
            let surveyStation = this.#surveyStations.at(indexStation);
            surveyStation.stationY = stationY;
        }
    }        

    /**
     * Gets the coordinate Z of the station with the specified index in metters
     * @param {number} indexStation 
     * @returns {string}
     */
    getStationZ(indexStation) {
        if(indexStation >= this.#surveyStations.length) {
            return null;
        }
        let surveyStation = this.#surveyStations.at(indexStation);
        return surveyStation.stationZ
    }

    /**
     * Saves the specified coordinate Z for station with the specified index in metters
     * @param {number} indexStation 
     * @param {string} stationZ 
     */
    saveStationZ(indexStation, stationZ) {
        if(indexStation < this.#surveyStations.length) {
            let surveyStation = this.#surveyStations.at(indexStation);
            surveyStation.stationZ = stationZ;
        }
    }            

    /**
     * Gets the height of the tool above the survey station-point for station 
     * with the specified index in metters
     * @param {number} indexStation 
     * @returns {string}
     */
    getStationHeight(indexStation) {
        if(indexStation >= this.#surveyStations.length) {
            return null;
        }
        let surveyStation = this.#surveyStations.at(indexStation);
        return surveyStation.stationHeight;
    }

    /**
     * Saves the height of the tool above the survey station-point for station with the specified index in metters
     * @param {number} indexStation 
     * @param {string} stationHeight 
     */
    saveStationHeight(indexStation, stationHeight) {
        if(indexStation < this.#surveyStations.length) {
            let surveyStation = this.#surveyStations.at(indexStation);
            surveyStation.stationHeight = stationHeight;
        }
    }    
    
    /**
     * Gets the specified direction to the landmark for the station 
     * with the specified index in d.mmss format
     * @param {number} indexStation 
     * @returns {string}
     */
    getOrDirection(indexStation) {
        if(indexStation >= this.#surveyStations.length) {
            return null;
        }
        let surveyStation = this.#surveyStations.at(indexStation);
        return surveyStation.orDirection;
    }

    /**
     * Saves the specified direction to the landmark for the station with the specified index in d.mmss format
     * @param {number} indexStation 
     * @param {string} orDirection 
     */
    saveOrDirection(indexStation, orDirection) {
        if(indexStation < this.#surveyStations.length) {
            let surveyStation = this.#surveyStations.at(indexStation);
            surveyStation.orDirection = orDirection;
        }        
    }

    /**
     * Gets the specified name of the landmark for the station 
     * with the specified index in d.mmss format
     * @param {number} indexStation 
     * @returns {string}
     */
    getOrName(indexStation) {
        if(indexStation >= this.#surveyStations.length) {
            return null;
        }
        let surveyStation = this.#surveyStations.at(indexStation);
        return surveyStation.orName;
    }


    /**
     * Saves the specified landmark name for the station with the specified index
     * @param {number} indexStation 
     * @param {string} orName 
     */
    saveOrName(indexStation, orName) {
        if(indexStation < this.#surveyStations.length) {
            let surveyStation = this.#surveyStations.at(indexStation);
            surveyStation.orName = orName;
        }        
    }

    /**
     * Gets the specified coordinate X of the landmark for the station 
     * with the specified index in meters
     * @param {number} indexStation 
     * @returns {string}
     */
    getOrX(indexStation) {
        if(indexStation >= this.#surveyStations.length) {
            return null;
        }
        let surveyStation = this.#surveyStations.at(indexStation);
        return surveyStation.orX;
    }


    /**
     * Saves the specified landmark coordinate X for the station with the specified index in meters
     * @param {number} indexStation 
     * @param {string} orX 
     */
    saveOrX(indexStation, orX) {
        if(indexStation < this.#surveyStations.length) {
            let surveyStation = this.#surveyStations.at(indexStation);
            surveyStation.orX = orX;
        }        
    }

    /**
     * Gets the specified coordinate Y of the landmark for the station 
     * with the specified index in meters
     * @param {number} indexStation 
     * @returns {string}
     */
    getOrY(indexStation) {
        if(indexStation >= this.#surveyStations.length) {
            return null;
        }
        let surveyStation = this.#surveyStations.at(indexStation);
        return surveyStation.orY;
    }

    /**
     * Saves the specified landmark coordinate Y for the station 
     * with the specified index in metters
     * @param {number} indexStation 
     * @param {string} orY
     */
    saveOrY(indexStation, orY) {
        if(indexStation < this.#surveyStations.length) {
            let surveyStation = this.#surveyStations.at(indexStation);
            surveyStation.orY = orY;
        }        
    }

    /**
     * Gets the target name for an item in the measurement collection 
     * with the specified indexes of position of repository
     * @param {number} indexStation 
     * @param {number} indexMeasurement 
     * @returns {string}
     */
    getTargetName(indexStation, indexMeasurement) {
        if(indexStation >= this.#surveyStations.lenth) {
            return null;
        }

        let surveyStation = this.#surveyStations.at(indexStation);

        if(indexMeasurement >= surveyStation.measurements.length) {
            return null;
        }
        let measurement = surveyStation.measurements.at(indexMeasurement);
        return measurement.targetName;
    }

    /**
     * Saves the target name for an item in the measurement collection 
     * with the specified indexes of position of repository
     * @param {number} indexStation 
     * @param {number} indexMeasurement 
     * @param {string} targetName 
     */
    saveTargetName(indexStation, indexMeasurement, targetName) {
        if(indexStation < this.#surveyStations.length) {
            let surveyStation = this.#surveyStations.at(indexStation);
            if(indexMeasurement < surveyStation.measurements.length) {
                let measurement = surveyStation.measurements.at(indexMeasurement);
                measurement.targetName = targetName;
            }
        }                
    }

    /**
     * Gets the direction to target for an item in the measurement collection 
     * with the specified indexes of position of repository in d.mmss format
     * @param {number} indexStation 
     * @param {number} indexMeasurement 
     * @returns {string}
     */
    getTargetDirection(indexStation, indexMeasurement) {
        if(indexStation >= this.#surveyStations.lenth) {
            return null;
        }

        let surveyStation = this.#surveyStations.at(indexStation);

        if(indexMeasurement >= surveyStation.measurements.length) {
            return null;
        }
        let measurement = surveyStation.measurements.at(indexMeasurement);
        return measurement.targetDirection;
    }
    
    /**
     * Saves the direction to target for an item in the measurement collection 
     * with the specified indexes of position of repository in d.mmss format
     * @param {number} indexStation 
     * @param {number} indexMeasurement 
     * @param {string} targetDirection 
     */
    saveTargetDirection(indexStation, indexMeasurement, targetDirection) {
        if(indexStation < this.#surveyStations.length) {
            let surveyStation = this.#surveyStations.at(indexStation);
            if(indexMeasurement < surveyStation.measurements.length) {
                let measurement = surveyStation.measurements.at(indexMeasurement);
                measurement.targetDirection = targetDirection;
            }
        }                
    }    

    /**
     * Gets the inclined distance to target for an item in the measurement collection 
     * with the specified indexes of position of repository in meters
     * @param {number} indexStation 
     * @param {number} indexMeasurement 
     * @returns {string}
     */
    getTargetDistance(indexStation, indexMeasurement) {
        if(indexStation >= this.#surveyStations.lenth) {
            return null;
        }

        let surveyStation = this.#surveyStations.at(indexStation);

        if(indexMeasurement >= surveyStation.measurements.length) {
            return null;
        }
        let measurement = surveyStation.measurements.at(indexMeasurement);
        return measurement.targetDistance;
    }


    /**
     * Saves the inclined distance to target for an item in the measurement collection 
     * with the specified indexes of position of repository in meters
     * @param {number} indexStation 
     * @param {number} indexMeasurement 
     * @param {string} targetDistance 
     */
    saveTargetDistance(indexStation, indexMeasurement, targetDistance) {
        if(indexStation < this.#surveyStations.length) {
            let surveyStation = this.#surveyStations.at(indexStation);
            if(indexMeasurement < surveyStation.measurements.length) {
                let measurement = surveyStation.measurements.at(indexMeasurement);
                measurement.targetDistance = targetDistance;
            }
        }                
    }    

    /**
     * Gets the tilt angle to target for an item in the measurement collection 
     * with the specified indexes of position of repository in d.mmss format
     * @param {number} indexStation 
     * @param {number} indexMeasurement 
     * @returns {string}
     */
    getTargetTiltAngle(indexStation, indexMeasurement) {
        if(indexStation >= this.#surveyStations.lenth) {
            return null;
        }

        let surveyStation = this.#surveyStations.at(indexStation);

        if(indexMeasurement >= surveyStation.measurements.length) {
            return null;
        }
        let measurement = surveyStation.measurements.at(indexMeasurement);
        return measurement.targetTiltAngle;
    }


    /**
     * Saves the tilt algle to target for an item in the measurement collection 
     * with the specified indexes of position of repository in d.mmss format
     * @param {number} indexStation 
     * @param {number} indexMeasurement 
     * @param {string} targetTiltAngle 
     */
    saveTargetTiltAngle(indexStation, indexMeasurement, targetTiltAngle) {
        if(indexStation < this.#surveyStations.length) {
            let surveyStation = this.#surveyStations.at(indexStation);
            if(indexMeasurement < surveyStation.measurements.length) {
                let measurement = surveyStation.measurements.at(indexMeasurement);
                measurement.targetTiltAngle = targetTiltAngle;
            }
        }                
    }    

    /**
     * Gets the height of the tool above the target-point for station 
     * with the specified index in meters
     * @param {number} indexStation 
     * @param {number} indexMeasurement 
     * @returns {string}
     */
    getTargetHeight(indexStation, indexMeasurement) {
        if(indexStation >= this.#surveyStations.lenth) {
            return null;
        }

        let surveyStation = this.#surveyStations.at(indexStation);

        if(indexMeasurement >= surveyStation.measurements.length) {
            return null;
        }
        let measurement = surveyStation.measurements.at(indexMeasurement);
        return measurement.targetHeight;
    }
    
    /**
     * Saves the height of the tool above the target-point for station 
     * with the specified index in meters
     * @param {number} indexStation 
     * @param {number} indexMeasurement 
     * @param {string} targetHeight 
     */
    saveTargetHeight(indexStation, indexMeasurement, targetHeight) {
        if(indexStation < this.#surveyStations.length) {
            let surveyStation = this.#surveyStations.at(indexStation);
            if(indexMeasurement < surveyStation.measurements.length) {
                let measurement = surveyStation.measurements.at(indexMeasurement);
                measurement.targetHeight = targetHeight;
            }
        }                
    }    


}

File: geocalculator_service/src/main/java/ru/taheoport/geocalculator_service/repository/SurveyRepository.java
package ru.taheoport.geocalculator_service.repository;

import ru.taheoport.geocalculator_service.model.Measurement;
import ru.taheoport.geocalculator_service.model.SurveyStation;

/**
 * Provides methods for the survey station repository
 */
public interface SurveyRepository {

    /**
     * Clears survey station repository
     */
    void clearAll();

    /**
     * Gets number of survey stations
     * @return int
     */
    int size();

    /**
     * Gets number of measurements on specified station index
     * @param stationIndex int station index
     * @return int number of measurements
     */
    int measurementSize(int stationIndex);

    /**
     * Adds new survey station to repository
     * @return SurveyStation The instance that was added
     */
    SurveyStation addNewStation();

    /**
     * Adds new measurement to station with specified index
     * @param stationIndex int
     * @return instance of Measurement that was added
     */
    Measurement addNewMeasurement(int stationIndex);

    /**
     * Gets name of station with specified index
     * @param stationIndex int
     * @return string
     */
    String getStationName(int stationIndex);

    /**
     * Saves name of station with specified index
     * @param stationIndex int
     * @param stationName string
     */
    void saveStationName(int stationIndex, String stationName);

    /**
     * Gets coordinate X in meters of station with specified index
     * @param stationIndex int
     * @return double coordinate X
     */
    long getStationX(int stationIndex);

    /**
     * Saves coordinate X in meters to station with specified index
     * @param stationIndex int
     * @param stationX double coordinate X
     */
    void saveStationX(int stationIndex, long stationX);

    /**
     * Gets coordinate Y in meters of station with specified index
     * @param stationIndex int
     * @return long coordinate Y
     */
    long getStationY(int stationIndex);

    /**
     * Saves coordinate Y in meters to station with specified index
     * @param stationIndex int
     * @param stationY long coordinate Y
     */
    void saveStationY(int stationIndex, long stationY);

    /**
     * Gets coordinate Z in meters of station with specified index
     * @param stationIndex int
     * @return long coordinate Z
     */
    long getStationZ(int stationIndex);

    /**
     * Saves coordinate Z in meters to station with specified index
     * @param stationIndex int
     * @param stationZ long coordinate Z
     */
    void saveStationZ(int stationIndex, long stationZ);

    /**
     * Gets height of the tool over the point in meters of
     * station with specified index
     * @param stationIndex int
     * @return long
     */
    long getStationHeight(int stationIndex);

    /**
     * Saves height of the tool over the point in meters to
     * station with specified index
     * @param stationIndex int
     * @param stationHeight long
     */
    void saveStationHeight(int stationIndex, long stationHeight);

    /**
     * Gets direction to back point (landmark) in radians of
     * station with specified index
     * @param stationIndex int
     * @return long
     */
    long getOrDirection(int stationIndex);

    /**
     * Saves direction to back point (landmark) in radians to
     * station with specified index
     * @param stationIndex int
     * @param orDirection long
     */
    void saveOrDirection(int stationIndex, long orDirection);

    /**
     * Gets name of the station with specified index
     * @param stationIndex int
     * @return String
     */
    String getOrName(int stationIndex);

    /**
     * Saves name to station with specified index
     * @param stationIndex int
     * @param orName String
     */
    void saveOrName(int stationIndex, String orName);

    /**
     * Gets coordinate X in meters of back point (landmark)
     * of station with specified index
     * @param stationIndex int
     * @return long coordinate X of landmark
     */
    long getOrX(int stationIndex);

    /**
     * Saves coordinate X in meters of back point (landmark)
     * to station with specified index
     * @param stationIndex int
     * @param  orX long
     */
    void saveOrX(int stationIndex, long orX);

    /**
     * Gets coordinate Y in meters of back point (landmark)
     * of station with specified index
     * @param stationIndex int
     * @return long coordinate Y of landmark
     */
    long getOrY(int stationIndex);

    /**
     * Saves coordinate Y in meters of back point (landmark)
     * to station with specified index
     * @param stationIndex int
     * @param  orY long
     */
    void saveOrY(int stationIndex, long orY);

    /**
     * Gets direction angle of baseline in radians of
     * station with specified index
     * @param stationIndex int
     * @return long
     */
    long getBaseDirectionAngle(int stationIndex);

    /**
     * Saves direction angle of baseline in radians to
     * station with specified index
     * @param stationIndex int
     * @param baseDirectionAngle long
     */
    void saveBaseDirectionAngle(int stationIndex, long baseDirectionAngle);

    /**
     * Gets name of target with specified index in measurements collection
     * of survey station with specified index
     * @param stationIndex int
     * @param targetIndex int
     * @return String
     */
    String getTargetName(int stationIndex, int targetIndex);

    /**
     * Saves name of target with specified index in measurements collection
     * of survey station with specified index
     * @param stationIndex int
     * @param targetIndex int
     * @param targetName String
     */
    void saveTargetName(int stationIndex, int targetIndex, String targetName);

    /**
     * Gets direction from station to this target in radians
     * with specified indexes of station and target
     * @param stationIndex int
     * @param targetIndex int
     * @return long
     */
    long getTargetDirection(int stationIndex, int targetIndex);

    /**
     * Saves direction from station to this target in radians
     * with specified indexes of station and target
     * @param stationIndex int
     * @param targetIndex int
     * @param targetDirection long
     */
    void saveTargetDirection(int stationIndex, int targetIndex, long targetDirection);

    /**
     * Gets inclined distance from station to this target in meters
     * with specified indexes of station and target
     * @param stationIndex int
     * @param targetIndex int
     * @return long
     */
    long getTargetInclinedDistance(int stationIndex, int targetIndex);

    /**
     * Saves inclined distance from station to this target in meters
     * with specified indexes of station and target
     * @param stationIndex int
     * @param targetIndex int
     * @param targetInclinedDistance long
     */
    void saveTargetInclinedDistance(int stationIndex, int targetIndex, long targetInclinedDistance);

    /**
     * Gets tilt angle from station to this target in radians
     * with specified indexes of station and target
     * @param stationIndex int
     * @param targetIndex int
     * @return long
     */
    long getTargetTiltAngle(int stationIndex, int targetIndex);

    /**
     * Saves tilt angle from station to this target in radians
     * with specified indexes of station and target
     * @param stationIndex int
     * @param targetIndex int
     * @param targetTiltAngle long
     */
    void saveTargetTiltAngle(int stationIndex, int targetIndex, long targetTiltAngle);

    /**
     * Gets height of the reflector over the target in meters
     * with specified indexes of station and target
     * @param stationIndex int
     * @param targetIndex int
     * @return long
     */
    long getTargetHeight(int stationIndex, int targetIndex);

    /**
     * Saves height of the reflector over the target in meters
     * with specified indexes of station and target
     * @param stationIndex int
     * @param targetIndex int
     * @param targetHeight long
     */
    void saveTargetHeight(int stationIndex, int targetIndex, long targetHeight);

    /**
     * Gets horizontal distance from station to target in meters
     * with specified indexes of station and target
     * @param stationIndex int
     * @param targetIndex int
     * @return long
     */
    long getTargetHorizontalDistance(int stationIndex, int targetIndex);

    /**
     * Saves horizontal distance from station to target in meters
     * with specified indexes of station and target
     * @param stationIndex int
     * @param targetIndex int
     * @param targetHorizontalDistance long
     */
    void saveTargetHorizontalDistance(int stationIndex, int targetIndex, long targetHorizontalDistance);

    /**
     * Gets direction angle from station to target in radians
     * with specified indexes of station and target
     * @param stationIndex int
     * @param targetIndex int
     * @return long
     */
    long getTargetDirectionAngle(int stationIndex, int targetIndex);

    /**
     * Saves direction angle from station to target in radians
     * with specified indexes of station and target
     * @param stationIndex int
     * @param targetIndex int
     * @param targetDirectionAngle long
     */
    void saveTargetDirectionAngle(int stationIndex, int targetIndex, long targetDirectionAngle);

    /**
     * Gets the X-axis coordinate increment between the station and target in meters
     * with specified indexes of station and target
     * @param stationIndex int
     * @param targetIndex int
     * @return long
     */
    long getTargetDeltaX(int stationIndex, int targetIndex);

    /**
     * Saves the X-axis coordinate increment between the station and target in meters
     * with specified indexes of station and target
     * @param stationIndex int
     * @param targetIndex int
     * @param targetDeltaX long
     */
    void saveTargetDeltaX(int stationIndex, int targetIndex, long targetDeltaX);

    /**
     * Gets the Y-axis coordinate increment between the station and target in meters
     * with specified indexes of station and target
     * @param stationIndex int
     * @param targetIndex int
     * @return long
     */
    long getTargetDeltaY(int stationIndex, int targetIndex);

    /**
     * Saves the Y-axis coordinate increment between the station and target in meters
     * with specified indexes of station and target
     * @param stationIndex int
     * @param targetIndex int
     * @param targetDeltaY long
     */
    void saveTargetDeltaY(int stationIndex, int targetIndex, long targetDeltaY);

    /**
     * Gets the Z-axis coordinate increment between the station and target in meters
     * with specified indexes of station and target
     * @param stationIndex int
     * @param targetIndex int
     * @return long
     */
    long getTargetDeltaZ(int stationIndex, int targetIndex);

    /**
     * Saves the Z-axis coordinate increment between the station and target in meters
     * with specified indexes of station and target
     * @param stationIndex int
     * @param targetIndex int
     * @param targetDeltaZ long
     */
    void saveTargetDeltaZ(int stationIndex, int targetIndex, long targetDeltaZ);

    /**
     * Gets coordinate X of target in meters
     * with specified indexes of station and target
     * @param stationIndex int
     * @param targetIndex int
     * @return long
     */
    long getTargetX(int stationIndex, int targetIndex);

    /**
     * Saves coordinate X of target in meters
     * with specified indexes of station and target
     * @param stationIndex int
     * @param targetIndex int
     * @param targetX long
     */
    void saveTargetX(int stationIndex, int targetIndex, long targetX);

    /**
     * Gets coordinate Y of target in meters
     * with specified indexes of station and target
     * @param stationIndex int
     * @param targetIndex int
     * @return long
     */
    long getTargetY(int stationIndex, int targetIndex);

    /**
     * Saves coordinate Y of target in meters
     * with specified indexes of station and target
     * @param stationIndex int
     * @param targetIndex int
     * @param targetY long
     */
    void saveTargetY(int stationIndex, int targetIndex, long targetY);

    /**
     * Gets coordinate Z of target in meters
     * with specified indexes of station and target
     * @param stationIndex int
     * @param targetIndex int
     * @return long
     */
    long getTargetZ(int stationIndex, int targetIndex);

    /**
     * Saves coordinate Z of target in meters
     * with specified indexes of station and target
     * @param stationIndex int
     * @param targetIndex int
     * @param targetZ long
     */
    void saveTargetZ(int stationIndex, int targetIndex, long targetZ);



}
